src/main/java/core/basesyntax/MainApp.java

```diff
@@ -1,5 +1,51 @@
 package core.basesyntax;
+public abstract class Machine {
+  public abstract void doWork();
+  public abstract void stopWork();
+}
 
-public class MainApp {
+public class Truck extends Machine {
+  @Override
+  public void doWork() {
+    System.out.println("Truck started working.");
+}
+  @Override
+  public void stopWork() {
+    System.out.println("Truck stopped working.");
+    } 
+}
 
+public class Bulldozer extends Machine {
+  @Override
+  public void doWork() {
+    System.out.println("Bulldozer started working.");
+  }
+  @Override
+  public void stopWork() {
+    System.out.println("Bulldozer stopped working.");
+    }
```

Comment:
Please don't add redundant empty lines to your code. There is an unnecessary empty line after the closing brace of the stopWork method in the Bulldozer class.